packages/core/src/services/issues/merge.test.ts

```diff
@@ -0,0 +1,163 @@
+      histograms: [
+        {
+          commitId: commit.id,
+          date: new Date('2024-01-01'),
+          count: 2,
+        },
+      ],
+    })
+
+    const now = new Date('2024-02-01')
+    await database
+      .update(issues)
+      .set({
+        centroid: { base: [1, 0], weight: 1 },
+        updatedAt: now,
+      })
+      .where(eq(issues.id, anchor.id))
+    await database
+      .update(issues)
+      .set({
+        centroid: { base: [0.9, 0.1], weight: 1 },
+        updatedAt: now,
+      })
+      .where(eq(issues.id, other.id))
+
+    const evaluation = await createEvaluationV2({
+      document,
+      commit,
+      workspace,
+      type: EvaluationType.Rule,
+      metric: RuleEvaluationMetric.ExactMatch,
+    })
+    const span = await createSpan({
+      workspaceId: workspace.id,
+      documentUuid: document.documentUuid,
+      commitUuid: commit.uuid,
+      type: SpanType.Prompt,
+    })
+    const result = await createEvaluationResultV2({
+      evaluation,
+      span,
+      commit,
+      workspace,
+      hasPassed: false,
+    })
+
+    await addIssueEvaluationResult({
+      issue: other,
+      evaluationResult: result,
+      workspaceId: workspace.id,
+    })
+    await database
+      .update(evaluationResultsV2)
+      .set({ issueId: other.id })
+      .where(eq(evaluationResultsV2.id, result.id))
+    await database
+      .update(evaluationVersions)
+      .set({ issueId: other.id })
+      .where(eq(evaluationVersions.id, evaluation.versionId))
+
+    const merging = await mergeIssues({ workspace, issue: anchor })
+    expect(merging.error).toBeFalsy()
+    const { winner, mergedIssues } = merging.unwrap()
+
+    expect(winner.id).toBe(anchor.id)
+    expect(mergedIssues.map((i) => i.id)).toContain(other.id)
+
+    const issuesRepository = new IssuesRepository(workspace.id)
+    const mergedIssue = await issuesRepository.find(other.id).then((r) =>
+      r.unwrap(),
+    )
+    expect(mergedIssue.mergedAt).not.toBeNull()
+
+    const histogramTotals = await database
+      .select({
+        total: sql<number>`SUM(${issueHistograms.count})`.as('total'),
+      })
+      .from(issueHistograms)
+      .where(eq(issueHistograms.issueId, winner.id))
+      .then((r) => r[0]!.total)
+    expect(histogramTotals).toBe(7)
+
+    const association = await database
+      .select()
+      .from(issueEvaluationResults)
+      .where(eq(issueEvaluationResults.issueId, winner.id))
+    expect(association.length).toBeGreaterThan(0)
+
+    const updatedResult = await database
+      .select()
+      .from(evaluationResultsV2)
+      .where(eq(evaluationResultsV2.id, result.id))
+      .then((r) => r[0]!)
+    expect(updatedResult.issueId).toBe(winner.id)
+
+    const updatedEvaluation = await database
+      .select()
+      .from(evaluationVersions)
+      .where(eq(evaluationVersions.id, evaluation.versionId))
+      .then((r) => r[0]!)
+    expect(updatedEvaluation.issueId).toBe(winner.id)
```

Comment:
Not testing 
1. centroid is updated
2. Eval results are assigned
3. issues are marked as merged
4. Vector DB is called with the right params

What else?